File: tkpdinbox/src/main/java/com/tokopedia/inbox/attachinvoice/di/AttachInvoiceComponent.java
package com.tokopedia.inbox.attachinvoice.di;

import android.content.Context;

import com.tokopedia.abstraction.common.di.component.BaseAppComponent;
import com.tokopedia.abstraction.common.di.qualifier.ApplicationContext;
import com.tokopedia.inbox.attachinvoice.view.fragment.AttachInvoiceFragment;

import dagger.Component;

/**
 * Created by Hendri on 05/04/18.
 */

@AttachInvoiceScope
@Component(modules = {AttachInvoiceModule.class}, dependencies = BaseAppComponent.class)
public interface AttachInvoiceComponent {

    void inject(AttachInvoiceFragment fragment);

    @ApplicationContext
    Context context();
}

File: features/user/attachinvoice/src/main/java/com/tokopedia/attachinvoice/di/AttachInvoiceComponent.kt
package com.tokopedia.attachinvoice.di

import com.tokopedia.abstraction.common.di.component.BaseAppComponent
import com.tokopedia.attachinvoice.view.fragment.AttachInvoiceFragment
import dagger.Component

@AttachInvoiceScope
@Component(
        modules = [AttachInvoiceModule::class, AttachInvoiceViewModelModule::class],
        dependencies = [BaseAppComponent::class]
)
interface AttachInvoiceComponent {
    fun inject(fragment: AttachInvoiceFragment)
}